src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/rc_helpers_test.cc

```diff
@@ -0,0 +1,278 @@
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the Ford Motor Company nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE. */
+
+#include "rc_rpc_plugin/rc_helpers.h"
+#include <map>
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <vector>
+#include "gtest/gtest.h"
+#include "rc_rpc_plugin/rc_module_constants.h"
+
+namespace {
+const std::string kKeyId = "id";
+const std::string kKeyStatus = "status";
+const std::string kKeyDensity = "density";
+const std::string kKeyValue = "value";
+const std::string kArray = "array";
+}  // namespace
+
+namespace rc_rpc_plugin_test {
+using namespace rc_rpc_plugin;
+
+class RCHelpersTest : public ::testing::Test {
+ public:
+  RCHelpersTest() {}
+
+  smart_objects::SmartObject genSmartObject(const int& object_number,
+                                            const bool& use_id_key) {
+    smart_objects::SmartObject result(smart_objects::SmartType_Map);
+    result[kKeyStatus] = kKeyStatus + std::to_string(object_number);
+    result[kKeyValue] = kKeyValue + std::to_string(object_number);
+    if (use_id_key) {
+      result[kKeyId] = kKeyId + std::to_string(object_number);
+    }
+    return result;
+  }
+};
+
+TEST_F(RCHelpersTest, MergeModuleData_OneOfParamsNotSmartMap) {
+  using namespace smart_objects;
+
+  SmartObject data1(SmartType_Map);
+  SmartObject data2(SmartType_Array);
+
+  auto result = RCHelpers::MergeModuleData(data1, data2);
+  EXPECT_NE(result, data1);
+  EXPECT_EQ(result, data2);
+}
+
+TEST_F(RCHelpersTest, MergeModuleData_FirstObjectDontContainKey) {
+  using namespace smart_objects;
+
+  SmartObject data1(SmartType_Map);
+  SmartObject data2(SmartType_Map);
+
+  std::map<std::string, std::string> expected_keys_values = {
+      {"key1", "value1"}, {"key2", "value2"}};
+  auto it = expected_keys_values.begin();
+
+  data1[it->first] = it->second;
+  ++it;
+  data2[it->first] = it->second;
+
+  auto result = RCHelpers::MergeModuleData(data1, data2);
+  for (auto& e : expected_keys_values) {
```

Comment:
@YarikMamykin same as above

Reply:
@AByzhynar , fixed in b61ba7c099f2c1b7caf78e8180514ad3e9cf253a